src/ldp/agent/simple_agent.py

```diff
@@ -71,6 +77,16 @@ def get_next_state(
             if hide_old_env_states is not None
             else self.hide_old_env_states
         )
+        if self.sliding_window is not None and self.sliding_window > 0:
+            msg_blocks = split_message_transitions(old_messages)
+            old_messages = (
+                msg_blocks[0]  # keep system messages + user message
```

Comment:
Won't this hit an `IndexError` if `old_messages` is empty? Also, feel free to ignore this one, if it won't happen